test: helm chart config map parsing

tommyknows opened the pull request:
This commit changes the existing "RealAPI" config test from using an inline config to using the config that we template in the Helm chart.

To achieve this, we introduce a "TemplateChart" function that templates a Helm chart at a given path with some values and returns the rendered manifests. We then extract the ConfigMap from there and write it to a file, which we then run our tests on.

The test assures that some default types from the Kubernetes APIs are present. It does not test any CRDs, as we don't register any and as such the GVK-lookup would fail.

Commit:
test: helm chart config map parsing

This commit changes the existing "RealAPI" config test from using an
inline config to using the config that we template in the Helm chart.

To achieve this, we introduce a "TemplateChart" function that templates
a Helm chart at a given path with some values and returns the rendered
manifests. We then extract the ConfigMap from there and write it to a
file, which we then run our tests on.

The test assures that some default types from the Kubernetes APIs are
present. It does not test any CRDs, as we don't register any and as such
the GVK-lookup would fail.

srlk commented:
seems fine to me but, it needs a `go mod tidy`

```
--- a/go.mod
+++ b/go.mod
@@ -13,6 +13,7 @@ require (
        k8s.io/api v0.26.1
        k8s.io/apimachinery v0.26.1
        k8s.io/client-go v0.26.1
+       k8s.io/kubectl v0.26.0
        sigs.k8s.io/controller-runtime v0.14.4
        sigs.k8s.io/yaml v1.3.0
```

Commit:
fix: service account Helm chart rendering

The go template would have removed both the newline above and below it,
which resulted in the serviceaccount's APIVersion being commented out.

tommyknows commented:
lol the test failed again because there's an actual rendering issue with the chart :smile: 

@srlk could you have another look?

snyk-deployer commented:
:tada: This PR is included in version 0.5.4 :tada:

The release is available on [GitHub release](https://github.com/snyk/kubernetes-scanner/releases/tag/v0.5.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
